server/src-lib/Data/Parser/JSONPath.hs

```diff
@@ -0,0 +1,51 @@
+module Data.Parser.JSONPath
+  ( parseJSONPath
+  , JSONPathElement(..)
+  ) where
+
+import           Control.Applicative  ((<|>))
+import           Data.Attoparsec.Text
+import           Prelude
+
+import qualified Data.Text            as T
+
+data JSONPathElement
```

Comment:
Why new `JSONPathElement` type since it is already defined in `aeson` library?
I suggest you define a parser for the type present in `aeson`.

Reply:
My idea is avoiding dependency. I will change it